Source/GSXibKeyedUnarchiver.m

```diff
@@ -143,8 +165,46 @@ - (NSData *) _preProcessXib: (NSData *)data
                       id key = [[xmlKeys objectAtIndex:index] stringValue];
                       if ([key rangeOfString:@"CustomClassName"].location != NSNotFound)
                         {
-                          // NSString *obj = [[xmlObjs objectAtIndex:index] stringValue];
-                          [customClassDict setObject:[[xmlObjs objectAtIndex:index] stringValue] forKey:key];
+                          NSString *cn = [[xmlObjs objectAtIndex: index] stringValue]; // className
+
+                          [customClassDict setObject:cn forKey:key];
+                                                    
+                          //
+                          // If we are in IB/Gorm build the custom classes map so that we don't instantiate
+                          // classes which don't exist (yet) in IB/Gorm.  This allows editing of the model
+                          // in IB/Gorm.  If we are in the live app, don't bother as it's a waste of memory.
+                          //
+                          if ([NSClassSwapper isInInterfaceBuilder] == YES)
+                            {
+                              NSString *xpath = [NSString stringWithFormat: @"//object[@class=\"IBClassDescriber\"]"
+                                                          @"//string[@key=\"className\"][text()=\"%@\"]"
+                                                          @"/../string[@key=\"superclassName\"]", cn];
+                              NSArray *descriptionObjs = [document nodesForXPath: xpath error: NULL];
+                              if ([descriptionObjs count] > 0)
+                                {
+                                  NSUInteger idx = [key rangeOfString: @"."].location;
+                                  if (idx != NSNotFound) // unlikely to be NSNotFound...
```

Comment:
Why don't you evaluate this before doing the xpath request?

Reply:
Done